vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -2299,7 +2301,13 @@ private int getLastFooterLayerIndex() {
         return -1;
     }
 
+    /**
+     * @inheritDocs
+     * @deprecated use instead one of the setDataSource methods from
+     * {@link HasListDataView} or {@link HasLazyDataView}
+     */
     @Override
+    @Deprecated
     public void setDataProvider(DataProvider<T, ?> dataProvider) {
         Objects.requireNonNull(dataProvider, "data provider cannot be null");
```

Comment:
Should we move the implementation to a `setDataSource` method that is not deprecated and call that one from here ?

Reply:
Perhaps not a setDataSource method, but perhaps some new private method that should be used.